### PR TITLE
etcdserver: include IsLeader in etcdserverpb.Member

### PR DESCRIPTION
**Pull request by gyuho**
This is especially useful to functional-test leader failure.

For https://github.com/coreos/etcd/issues/4477.

### CODE REVIEW

**Comment by gyuho** [comment]
/cc @xiang90 @heyitsanthony 

Thanks.

**Comment by heyitsanthony** [comment]
would this be better as a flag in `message Member` instead of having a dedicated function?

File: etcdserver/api/v3rpc/member.go [hunk 1]
```diff
@@ -95,6 +95,18 @@ func (cs *ClusterServer) MemberUpdate(ctx context.Context, r *pb.MemberUpdateReq
 	return &pb.MemberUpdateResponse{Header: cs.header()}, nil
 }
 
+func (cs *ClusterServer) MemberLeader(ctx context.Context, r *pb.MemberLeaderRequest) (*pb.MemberLeaderResponse, error) {
+	membs := cs.cluster.Members()
+	var peerURLs []string
+	if len(membs) > 0 {
+		peerURLs = membs[0].PeerURLs
```
Comment (heyitsanthony) [inline etcdserver/api/v3rpc/member.go]: is `membs[0]` always the leader?

Comment (gyuho) [inline etcdserver/api/v3rpc/member.go]: It's sorted by ID, so I think it's not always (https://github.com/coreos/etcd/blob/master/etcdserver/member.go#L160-L165).

**Comment by gyuho** [comment]
@heyitsanthony I will try to include leader information in `Member` message and let get leader with `MemberList` method, instead of having this function.

Thanks.

**Comment by gyuho** [comment]
@heyitsanthony Just addressed. PTAL.Thanks!

File: etcdserver/etcdserverpb/rpc.proto [hunk 2]
```diff
@@ -323,10 +323,11 @@ message Member {
   uint64 ID = 1;
   // If the member is not started, name will be an empty string.
   string name = 2;
-  repeated string peerURLs = 3;
+  uint64 LeaderID = 3;
```
Comment (heyitsanthony) [inline etcdserver/etcdserverpb/rpc.proto]: `bool isLeader`?

Comment (gyuho) [inline etcdserver/etcdserverpb/rpc.proto]: Yeah boolean is easier to use. Will change.

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Just changed it to `IsLeader`. Thanks!

**Comment by xiang90** [comment]
can we add another API called memberLeader instead of adding this field?

**Comment by xiang90** [comment]
Or shall we do it in client library?

**Comment by gyuho** [comment]
@xiang90 I have code that adds `MemberLeader` but I think just having boolean value in `Member` and get it from `MemberList` feel simpler.

**Comment by heyitsanthony** [comment]
@xiang90 MemberLeader() can go in the client library.

**Comment by gyuho** [comment]
So `MemberLeader` in `clientv3`?

**Comment by heyitsanthony** [comment]
yeah, the clientv3 implementation would just process memberlist if it's using the bool flag

**Comment by gyuho** [comment]
Ok will add that. Thanks.

**Comment by gyuho** [comment]
Just added `MemberLeader` method. PTAL. Thanks!

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks. Will merge in green lights.

File: clientv3/cluster.go [hunk 3]
```diff
@@ -33,6 +34,9 @@ type Cluster interface {
 	// List lists the current cluster membership.
 	MemberList(ctx context.Context) (*MemberListResponse, error)
 
+	// Leader returns the current leader member.
```
Comment (heyitsanthony) [inline clientv3/cluster.go]: godoc?

Comment (gyuho) [inline clientv3/cluster.go]: I missed this comment. You mean to make the comment `MemberLeader`? I can change that in a separate PR with the fix for `MemberList` as well.

**Pull request merged**